From f79b7226a316508ae886dc87547ecf4469b76bee Mon Sep 17 00:00:00 2001
From: zamiseck <zimzam17@gmail.com>
Date: Fri, 25 Mar 2022 13:00:59 -0500
Subject: [PATCH] Add Error to Software Version Interface

This commit adds a InvalidSignature error to the
Software Version interface. It will ensure that
if the error is hit during the verificatioon of
an image, a corresponding error is available to the
user.

Signed-off-by: Zami Seck <zimzam17@gmail.com>
Change-Id: Iba2eb7689a3e34db15e4a5d31accb69c18f906f5
---
 yaml/xyz/openbmc_project/Software/Version.errors.yaml   | 3 +++
 yaml/xyz/openbmc_project/Software/Version.metadata.yaml | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/yaml/xyz/openbmc_project/Software/Version.errors.yaml b/yaml/xyz/openbmc_project/Software/Version.errors.yaml
index e62ee4b..9840b30 100644
--- a/yaml/xyz/openbmc_project/Software/Version.errors.yaml
+++ b/yaml/xyz/openbmc_project/Software/Version.errors.yaml
@@ -8,3 +8,6 @@
 - name: AlreadyExists
   description: >
     This image version already exists on the device
+- name: InvalidSignature
+  description: >
+    Signature Validation failed for image version.
diff --git a/yaml/xyz/openbmc_project/Software/Version.metadata.yaml b/yaml/xyz/openbmc_project/Software/Version.metadata.yaml
index a3d4c4f..6550d0e 100644
--- a/yaml/xyz/openbmc_project/Software/Version.metadata.yaml
+++ b/yaml/xyz/openbmc_project/Software/Version.metadata.yaml
@@ -12,3 +12,5 @@
   meta:
     - str: "IMAGE_VERSION=%s"
       type: string
+- name: InvalidSignature
+  level: ERR
-- 
2.25.1

